### DISCUSSION TITLE
Can we do a shake effect?

### DISCUSSION THREAD

**Discussion by alex285** [olive-editor/olive#2166] [General]
Is possible to create a shake effect from the transformation node, without manually keyframe everything?

**Comment by defini-tiv** [comment]
Yes:
```
Time --> Math(Multiply) --> Trigonometry(Sin) --> Math(Multiply)  <-- Value(Vector 2D(y))
 |                                                     ↓
 |                                                Math (Addition) --> Transform(Positon)
 |                                                     ↑
  -----> Math(Multiply) --> Trigonometry(Sin) --> Math(Multiply)  <--  Value(Vector 2D(x))
```
The multiplies are for speed control, the vectors for amplitude. One for x, one for y.  You can replace the sin function by a more complex one e.g. for more randomness or fade in/out.

**Reply by alex285** [reply]
is that even close? :/
![Screenshot from 2023-02-19 17-54-23](https://user-images.githubusercontent.com/11244308/219959282-67936990-8aeb-4464-ad09-c9ce1a760044.png)

**Reply by defini-tiv** [reply]
Yes, you just need to disconnect shape from time and connect it to the texture from transform.

**Reply by defini-tiv** [reply]
Oh and you need the sin function to become one value of the multiplication and not to enable it. You really don't need to turn off a any node in this use case. Just make sure all of them are statically enabled.
